pkg/controller/traceflow/controller.go

```diff
@@ -276,34 +282,25 @@ func (c *Controller) checkTraceflowStatus(tf *opsv1alpha1.Traceflow) (retry bool
 		return
 	}
 	if time.Now().UTC().Sub(tf.CreationTimestamp.UTC()).Seconds() > timeout {
-		_, err = c.errorTraceflowCRD(tf, "traceflow timeout")
+		err = c.errorTraceflowCRD(tf, "traceflow timeout")
 		return
 	}
 	retry = true
 	return
 }
 
-func (c *Controller) runningTraceflowCRD(tf *opsv1alpha1.Traceflow, dataPlaneTag uint8) (*opsv1alpha1.Traceflow, error) {
+func (c *Controller) runningTraceflowCRD(tf *opsv1alpha1.Traceflow, dataPlaneTag uint8) error {
 	tf.Status.DataplaneTag = dataPlaneTag
 	tf.Status.Phase = opsv1alpha1.Running
-
-	type Traceflow struct {
-		Status opsv1alpha1.TraceflowStatus `json:"status,omitempty"`
-	}
-	patchData := Traceflow{Status: opsv1alpha1.TraceflowStatus{Phase: tf.Status.Phase, DataplaneTag: dataPlaneTag}}
-	payloads, _ := json.Marshal(patchData)
-	return c.client.OpsV1alpha1().Traceflows().Patch(context.TODO(), tf.Name, types.MergePatchType, payloads, metav1.PatchOptions{}, "status")
+	_, err := c.client.OpsV1alpha1().Traceflows().UpdateStatus(context.TODO(), tf, metav1.UpdateOptions{})
```

Comment:
Can we use RetryOnConflict to catch this? Like what we did in agent.
https://github.com/vmware-tanzu/antrea/blob/880794f5589b641fd77cb60e9ca8b71d7202c69a/pkg/agent/controller/traceflow/packetin.go#L46

Reply:
I thought we will always retry at any error, and even patch can fail and we should retry too.

I can add retry, but what the problem to retry in processTraceflowItem() to follow the standard controller retry model, instead of adding a new retry mechanisms.

Reply:
I just checked the code, and feel we should always retry at processTraceflowItem() at any error of processTraceflowItem(), except the case the TF CRD is already deleted. Would you agree?

Reply:
I'm OK if the controller won't retry indefinitely.
Please note we need to retry periodically for a Running TF, to check if the work is done or timeout, the logic is in checkTraceflowStatus

Reply:
I actually feel controller should retry indefinitely for the current errors as they should be recoverable.
@tnqn : thoughts?

Reply:
Agree, infinite retries with a rate limit queue is a common pattern of controller. The NotFound error is usually handled after `tf, err := c.traceflowLister.Get(traceflowName)`  by not returning an error (then the item will be removed from the queue)
For the current errors, there seems no real reason to stop retrying: even for the error of no available tag, it could keep retrying until it timeouts. 
Maybe we could return a single error value to indicate whether it should be retried to be simpler.
AFAIK, both of retrying immediately and relying on workqueue to retry are used in K8s. I think the former way could avoid some repeated processing before the API call (in next rounds) and finish the task earlier while the latter processes the whole tasks in a more fair way by not blocking on any specific task (the retry comes with backoff). I don't have strong opinion on which mode should be used here.

Reply:
considering TF is a diagnostic tool for on demand purposes.. we may not need to have so many conflicts so relying on work queue for retries should be okay..

Reply:
Thanks for the review. I will update the PR to retry for all errors.